pkg/api/types.go

```diff
@@ -271,8 +271,16 @@ type KubernetesAddon struct {
 	Data       string                    `json:"data,omitempty"`
 }
 
-// IsEnabled returns if the addon is explicitly enabled, or the user-provided default if non explicitly enabled
-func (a *KubernetesAddon) IsEnabled(ifNil bool) bool {
+// IsEnabled returns if the addon is enabled
+func (a *KubernetesAddon) IsEnabled() bool {
+	if a.Enabled == nil {
+		return false
+	}
+	return *a.Enabled
+}
+
+// GetEnabledIfNil returns if the addon is explicitly enabled, or the user-provided default if non explicitly enabled
+func (a *KubernetesAddon) GetEnabledIfNil(ifNil bool) bool {
```

Comment:
I don't really understand the function name or its comment

Reply:
It's a new name for the previous "IsEnabled" method. Hold on discussing the name until I determine whether or not we can get rid of this foo altogether as I get closer to finishing this PR.